### PR TITLE
Switch to hakyll

### LINKED ISSUE #8: Hook up Hakyll again

### PR DESCRIPTION
**Pull request by hdgarrood**
Fixes #8 

There's a version of this up at http://harry.garrood.me/purescript-www-dev/. It should be the same, I think (apart from changes like #53 are now in the HTML).

There are just two things I'm aware of that this PR doesn't address:

* How to make the generated site visible to GitHub Pages. My preferred way to do this is to use a separate branch for the generated site; this way, the git history isn't full of changes to generated files.
* Created/updated dates. I wasn't sure what the creation dates of some of the posts were, because only the update date was listed, so I just changed the article headers to say "published on <date>". I could set it up to distinguish create/update dates fairly easily if I had this information.

Rather than merging this here, you might want to pull these changes into a separate branch manually, run the generator, and then check the generated site in to `master`. Perhaps we can try to streamline this later, eg, with Travis or something.

### CODE REVIEW

**Comment by AppShipIt** [comment]
Lol I didn't realize how impending this is great work 🙏 As for branch I suppose we could have a `generator` branch that's set as the default? My impression is the hosted content has to be in `master` using the `github.io` naming convention?

**Comment by hdgarrood** [comment]
Yeah, I think that's right.

**Comment by hdgarrood** [comment]
Ah, I've just realised the page titles need a bit of work. The homepage is now "PureScript - Home - index`. Give me a minute.

**Comment by AppShipIt** [comment]
> My impression is hosted content has to be in `master`

[Confirmed](https://help.github.com/articles/user-organization-and-project-pages/)

> There's also a git command for making one branch inside another.

Using [`subtree`](http://yeoman.io/learning/deployment.html). There are also `grunt` and `gulp` plugins for that.

**Comment by hdgarrood** [comment]
Ah, I wasn't aware of how you could use git subtree for that - that looks great!

**Comment by hdgarrood** [comment]
Oh, but it seems it does still require you to check the generated site in. That's not a huge deal, I guess.

File: templates/default.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      <ul class="external">
+        <li class="github"><a href="https://github.com/purescript/purescript" title="GitHub">GitHub</a></li>
+        <li class="twitter"><a href="https://twitter.com/purescript" title="Twitter">Twitter</a></li>
+      </ul>
+    </nav>
+  </footer>
+
+  <script src="js/lib/zepto-1.1.4.min.js"></script>
+  <script src="js/home.js"></script>
+  <script>
```
Comment (paf31) [inline templates/default.html]: This GA snippet can actually be removed now, I'm not tracking this anywhere.

**Comment by paf31** [comment]
Looks great, thanks!

My take on the JS vs Haskell thing is that Haskell does what we need for now, but I'd be willing to switch if we could use it as an excuse to create a static site generator in PureScript, which I think could be a really great application.

**Comment by AppShipIt** [comment]
Likewise now it's working I'm not nearly as motivated to undertake that short term 👍

**Comment by hdgarrood** [comment]
Yeah, agreed re aiming to switch to PureScript eventually. I feel the same way about Pursuit; I'd rather it were in PureScript, but realistically, it wouldn't have happened in time (that is, over the course of GSOC) had I tried to do in it PureScript. Creating a static site generator in PS sounds fun but I think we have a fair bit of lower hanging fruit right now. :)

I've removed the GA snippet.

**Comment by paf31** [comment]
:+1: This looks good to me. @garyb, good to merge?

It'll be nice to add RSS/Atom support once this is merged, then we can get blog posts syndicated on Planet Haskell etc.

**Comment by paf31** [comment]
@hdgarrood When I merge this, will the site still be available on `gh-pages`? What steps do I need to take the deploy this?

**Comment by paf31** [comment]
Oh, we're not using `gh-pages`.

**Comment by AppShipIt** [comment]
Ah let me take a look at automating this tonight 👍 I do recommend merging it into a `generator` branch and setting that as the repo default branch regardless. Just a side effect of not using `gh-pages` (not a bad thing).

**Comment by paf31** [comment]
Automating this would be great. I'll merge this into `generator`.

**Pull request closed without merging**